app/repositories/EditAddressLockRepository.scala

```diff
@@ -21,62 +21,103 @@ import java.time.{OffsetDateTime, ZoneId, ZoneOffset}
-class CorrespondenceAddressLockRepository @Inject()(mongo: ReactiveMongoApi, implicit val ec: ExecutionContext) {
+class EditAddressLockRepository @Inject()(
+  configDecorator: ConfigDecorator
+)(mongo: ReactiveMongoApi, implicit val ec: ExecutionContext) {
 
-  private val collectionName: String = "correspondenceAddressLock"
+  private val collectionName: String = "EditAddressLock"
   private val duplicateKeyErrorCode = "E11000"
 
   private def collection: Future[JSONCollection] =
     mongo.database.map(_.collection[JSONCollection](collectionName))
 
-  import CorrespondenceAddressLockRepository._
+  import EditAddressLockRepository._
 
-  def insert(nino: String): Future[Boolean] = {
-    val date = getNextMidnight(OffsetDateTime.now())
-    insertCore(nino, date).map(_.ok) recover {
-      case e: DatabaseException if e.getMessage().contains(duplicateKeyErrorCode) => false
+  def insert(nino: String, addressType: AddrType): Future[Boolean] = {
+
+    val record: EditedAddress =
+      AddrType.toEditedAddress(addressType, toBSONDateTime(getNextMidnight(OffsetDateTime.now())))
+
+    insertCore(AddressJourneyTTLModel(nino, record)).map(_.ok) recover {
+      case e: DatabaseException => false
     }
   }
 
-  def get(nino: String): Future[Option[AddressJourneyTTLModel]] =
+  def get(nino: String): Future[List[AddressJourneyTTLModel]] = {
+
+    val getExpiredAt = "editedAddress.expireAt"
+
     getCore(
       BSONDocument(
-        BSONDocument("_id"     -> nino),
-        BSONDocument(EXPIRE_AT -> BSONDocument("$gt" -> toBSONDateTime(OffsetDateTime.now()))))
+        BSONDocument("nino" -> nino),
+        BSONDocument(
+          BSONDocument(getExpiredAt -> BSONDocument("$gt" -> toBSONDateTime(OffsetDateTime.now())))
+        )
+      )
     )
+  }
 
-  private[repositories] def getCore[S](selector: BSONDocument): Future[Option[AddressJourneyTTLModel]] =
-    this.collection.flatMap(_.find(selector, None).one[AddressJourneyTTLModel])
-
-  private[repositories] def insertCore(nino: String, date: OffsetDateTime): Future[WriteResult] =
-    this.collection.flatMap(_.insert(ordered = false).one(AddressJourneyTTLModel(nino, toBSONDateTime(date))))
+  private[repositories] def getCore[S](selector: BSONDocument): Future[List[AddressJourneyTTLModel]] =
+    this.collection
+      .flatMap {
+        _.find(selector, None)
+          .cursor[AddressJourneyTTLModel]()
+          .collect[List](
+            5,
+            Cursor.FailOnError[List[AddressJourneyTTLModel]]()
+          )
+      }
+      .recover {
+        case e: Exception =>
+          Logger.error(s"Unable to find document: ${e.getMessage}")
+          List[AddressJourneyTTLModel]()
+      }
+
+  private[repositories] def insertCore(record: AddressJourneyTTLModel): Future[WriteResult] = {
+    val index = Index(Seq((record.createIndex, IndexType.Ascending)), unique = false)
+
+    this.collection.flatMap(coll =>
+      coll.indexesManager.ensure(index).flatMap { result =>
```

Comment:
You don't need to ensure the index on every insert, once when the application starts is enough, maybe put it in the `setIndex` method